integration/spark/src/main/scala/org/apache/spark/sql/parser/CarbonSpark2SqlParser.scala

```diff
@@ -766,13 +766,13 @@ class CarbonSpark2SqlParser extends CarbonDDLSqlParser {
       throw new MalformedCarbonCommandException("Invalid table properties")
     }
     if (options.isBucketingEnabled) {
-      if (options.bucketNumber.toString.contains("-") ||
-          options.bucketNumber.toString.contains("+") ||  options.bucketNumber == 0) {
+      if (options.bucketNumber.isEmpty || options.bucketNumber.get.toString.contains("-") ||
```

Comment:
Since you have wrapped options with Try, i guess in case if Bucket number as "+" and "-", it will be empty. Can check and remove those checks

Reply:
The check for "-" is needed to avoid negative values as input. The case for "+" is not required and removed the check for it.